File: src/main/java/com/paragon/client/systems/module/impl/misc/TimerModule.java
package com.paragon.client.systems.module.impl.misc;

import com.paragon.asm.mixins.accessor.IMinecraft;
import com.paragon.asm.mixins.accessor.ITimer;
import com.paragon.client.systems.module.Module;
import com.paragon.client.systems.module.Category;
import com.paragon.client.systems.module.setting.Setting;

/**
 * @author Wolfsurge
 */
public class TimerModule extends Module {

    public static TimerModule INSTANCE;

    public static Setting<Float> timer = new Setting<>("Timer Speed", 1.25f, 0.01f, 20f, 0.01f)
            .setDescription("How much to multiply the timer speed by");

    public TimerModule() {
        super("Timer", Category.MISC, "Modifies how long each tick takes");

        INSTANCE = this;
    }

    @Override
    public void onDisable() {
        ((ITimer) ((IMinecraft) mc).getTimer()).setTickLength(50);
    }

    @Override
    public void onTick() {
        if (nullCheck()) {
            return;
        }

        ((ITimer) ((IMinecraft) mc).getTimer()).setTickLength(50 / timer.getValue());
    }
}

File: src/main/kotlin/com/paragon/client/systems/module/impl/misc/TimerModule.kt
package com.paragon.client.systems.module.impl.misc

import com.paragon.api.module.Category
import com.paragon.api.module.Module
import com.paragon.api.setting.Setting
import com.paragon.api.util.anyNull
import com.paragon.mixins.accessor.IMinecraft
import com.paragon.mixins.accessor.ITimer

/**
 * @author Surge
 */
object TimerModule : Module("Timer", Category.MISC, "Modifies how long each tick takes") {

    private val timer = Setting(
        "TimerSpeed",
        1.25f,
        0.01f,
        4f,
        0.01f
    ) describedBy "How much to multiply the timer speed by"

    override fun onDisable() {
        ((minecraft as IMinecraft).timer as ITimer).tickLength = 50f
    }

    override fun onTick() {
        if (minecraft.anyNull) {
            return
        }

        ((minecraft as IMinecraft).timer as ITimer).tickLength = 50 / timer.value
    }

}